hazelcast/src/main/java/com/hazelcast/query/impl/QueryableEntry.java

```diff
@@ -46,14 +48,31 @@
     protected InternalSerializationService serializationService;
     protected Extractors extractors;
 
-    private Metadata metadata;
+    private StoreAdapter storeAdapter;
+    private Record record;
 
+    // overridden in some subclasses
     public Metadata getMetadata() {
-        return metadata;
+        if (record != null) {
```

Comment:
Is this null check here because of query cache indexes?

Reply:
And also because of some unit tests. I will double check, and at least document this.